intezer_sdk/analysis.py

```diff
@@ -66,26 +69,33 @@ def send(self, wait: typing.Union[bool, int] = False) -> None:
 
         if wait:
             if isinstance(wait, int):
-                self.wait_for_completion(wait, sleep_before_first_check=True)
+                self.wait_for_completion(wait, sleep_before_first_check=True, timeout=wait_timeout)
             else:
-                self.wait_for_completion(sleep_before_first_check=True)
+                self.wait_for_completion(sleep_before_first_check=True, timeout=wait_timeout)
 
-    def wait_for_completion(self, interval: int = None, sleep_before_first_check=False):
+    def wait_for_completion(self,
+                            interval: int = None,
+                            sleep_before_first_check=False,
+                            timeout: typing.Optional[datetime.timedelta] = None):
         """
         Blocks until the analysis is completed
         :param interval: The interval to wait between checks
         :param sleep_before_first_check: Whether to sleep before the first status check 
+        :param timeout: Maximum duration to wait for analysis completion
         """
+        start_time = datetime.datetime.now()
         if not interval:
             interval = consts.CHECK_STATUS_INTERVAL
         if self._is_analysis_running():
             if sleep_before_first_check:
                 time.sleep(interval)
             status_code = self.check_status()
 
-            while status_code != consts.AnalysisStatusCode.FINISH:
+            timeout_passed = timeout and datetime.datetime.now() - start_time > timeout
+            while not timeout_passed and status_code != consts.AnalysisStatusCode.FINISH:
                 time.sleep(interval)
                 status_code = self.check_status()
+                timeout_passed = timeout and datetime.datetime.now() - start_time > timeout
```

Comment:
i wonder if we should raise something or alert the user somehow? it feels that if someone set a timeout eh expects to "catch" it somehow,

Reply:
relevant also to all the other places that wait_timeout has been added, we should talk about the behavior

Reply:
raised TimeoutError on timeouts